### PR TITLE
Amp task2027 add log dir

### PR DESCRIPTION
**Pull request by gitpaulsmith**

### CODE REVIEW

File: oms/process_forecasts.py [hunk 1]
```diff
@@ -86,6 +79,13 @@ def _get_object_from_config(key: str, expected_type: type) -> Any:
     ath_end_time = _get_object_from_config("ath_end_time", datetime.time)
     trading_end_time = _get_object_from_config("trading_end_time", datetime.time)
     hdbg.dassert_lte(trading_end_time, ath_end_time)
+    #
+    execution_mode = _get_object_from_config("execution_mode", str)
+    #
+    if "log_dir" in config:
```
Comment (gpsaggese) [inline oms/process_forecasts.py:85]: FYI you can do

log_dir = config.get("log_dir", None)

File: oms/portfolio.py [hunk 2]
```diff
@@ -311,6 +313,29 @@ def get_historical_holdings_marked_to_market(self) -> pd.DataFrame:
         asset_values[AbstractPortfolio.CASH_ID] = cash
         return asset_values
 
+    def log_state(self, log_dir: str) -> None:
+        hdbg.dassert(log_dir, "Must specify `log_dir` to log state.")
+        #
+        wall_clock_time = self._get_wall_clock_time()
+        filename = str(wall_clock_time) + ".csv"
```
Comment (gpsaggese) [inline oms/portfolio.py:320]: maybe a `file_name = file_name.replace(" ", "_)` to avoid spaces in the name (which create problems)

**Review by gpsaggese — COMMENTED** [review]
Looks good. 2 nits

File: oms/process_forecasts.py (hunk 1 above)
Comment (gitpaulsmith) [reply oms/process_forecasts.py]: Done

File: oms/portfolio.py (hunk 2 above)
Comment (gitpaulsmith) [reply oms/portfolio.py:320]: Done

**Pull request merged**